File: packages/camera/camera_avfoundation/example/ios/RunnerTests/Mocks/MockFlutterTextureRegistry.m
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "MockFlutterTextureRegistry.h"

@implementation MockFlutterTextureRegistry

- (int64_t)registerTexture:(nonnull NSObject<FlutterTexture> *)texture {
  return 0;
}

- (void)textureFrameAvailable:(int64_t)textureId {
}

- (void)unregisterTexture:(int64_t)textureId {
}

@end

File: packages/camera/camera_avfoundation/darwin/Tests/Mocks/MockFlutterTextureRegistry.swift
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import Flutter

/// Mocked implementation of `FlutterTextureRegistry` protocol that exists to allow constructing
/// a `CameraPlugin` instance for testing. It contains an empty implementation for all protocol
/// methods.
final class MockFlutterTextureRegistry: NSObject, FlutterTextureRegistry {
  func register(_ texture: FlutterTexture) -> Int64 { 0 }

  func textureFrameAvailable(_ textureId: Int64) {}

  func unregisterTexture(_ textureId: Int64) {}
}
